CourseManager/ViewModels/CourseDetailsViewModel.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  CourseDetailsViewModel.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-10-04.
+//
+
+import FirebaseCore
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import Foundation
+import SwiftUI
+
+class CourseDetailsViewModel: ObservableObject {
+    @Published var course: CourseInformation = .init(id: "", name: "", url: "", gitHubURL: "", status: .todo, dateOfCompletion: Date(), comments: "")
+    private var db = Firestore.firestore()
+
+    func getCourseInformation(courseID: String) {
+        let courseRef = db.collection("courses").document(courseID)
```

Comment:
@sghourab you use `collection("courses")` across the app, make a singleton class which will keep all your collections and reuse the class across the app. 

Don't need to "copy-paste" hardcoded code across the app.

Reply:
I created the class below and deleted any (private var db = Firestore.firestore()) AND db.collection("courses) across the files: 

import FirebaseFirestore
import FirebaseFirestoreSwift

class CourseRepository: ObservableObject {
    static let db = Firestore.firestore()
    static let collection = db.collection("courses")
}